From 746e7c9efa37cec9a44d84a1e96b8c38f385cc1f Mon Sep 17 00:00:00 2001
From: "H. Peter Anvin (Intel)" <hpa@zytor.com>
Date: Sun, 12 Oct 2025 13:05:55 -0700
Subject: [PATCH] compiler.h: the test for "neither C++ nor C23" still wrong

The test needs to test for neither nor; as it was it tested "(not C++)
or C23" which was not at all what was intended...

Signed-off-by: H. Peter Anvin (Intel) <hpa@zytor.com>
---
 include/compiler.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/compiler.h b/include/compiler.h
index 53a9b5c1..d4cb7161 100644
--- a/include/compiler.h
+++ b/include/compiler.h
@@ -182,7 +182,7 @@ char * pure_func strrchrnul(const char *, int);
 #endif
 
 /* C++ and C23 have bool, false, and true as proper keywords */
-#if !defined(__cplusplus) || (__STDC_VERSION__ >= 202311L)
+#if !defined(__cplusplus) && (__STDC_VERSION__ < 202311L)
 # ifdef HAVE_STDBOOL_H
 #  include <stdbool.h>
 # elif defined(HAVE___BOOL)
